File: Remove leading zeros from an IP address - GFG/remove-leading-zeros-from-an-ip-address.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution
{
  public:
    string newIPAdd (string S)
    {
         
        string r="";
        int n=S.size();
        for(int i=0;i<n;i++){
            string t="";
            while(i<n && S[i]=='0')i++;
            while(i<n && S[i]!='.') t+=S[i++];
            if(t.size()==0)t='0';
            r+=i<n ? t+'.': t;
            
        }
        return r;
    }
};
//{ Driver Code Starts.
int main()
{
    int t; cin >> t;
    while (t--)
    {
        string s; cin >> s;
        Solution ob;
        cout <<ob.newIPAdd (s) << endl;
    }
}


// } Driver Code Ends

File: Remove leading zeros from an IP address - GFG/remove-leading-zeros-from-an-ip-address.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

public class GfG
{
    public static void main(String args[])
        {
            Scanner sc = new Scanner(System.in);
            int t = sc.nextInt();
            while(t-->0)
                {
                    String s = sc.next();
                    Solution ob = new Solution();
                    System.out.println(ob.newIPAdd(s));
                }
        }
}
// } Driver Code Ends


//User function Template for Java

class Solution
{
    public String newIPAdd(String S)
    {
    String r="";
        int n=S.length();
        for(int i=0;i<n;i++){
            String t="";
            while(i<n && S.charAt(i)=='0')i++;
            while(i<n && S.charAt(i)!='.') t+=S.charAt(i++);
            if(t.length()==0)t ="0";
            r+=i<n ? t+'.': t;
            
        }
        return r;
    }
    
}